codec-http2/src/main/java/io/netty/handler/codec/http2/StreamBufferingEncoder.java

```diff
@@ -69,33 +69,45 @@ public Http2ChannelClosedException() {
         }
     }
 
+    private static final class GoAwayDetail {
+        private final int lastStreamId;
+        private final long errorCode;
+        private final byte[] debugData;
+
+        GoAwayDetail(int lastStreamId, long errorCode, byte[] debugData) {
+            this.lastStreamId = lastStreamId;
+            this.errorCode = errorCode;
+            this.debugData = debugData.clone();
+        }
+    }
+
     /**
      * Thrown by {@link StreamBufferingEncoder} if buffered streams are terminated due to
      * receipt of a {@code GOAWAY}.
      */
     public static final class Http2GoAwayException extends Http2Exception {
         private static final long serialVersionUID = 1326785622777291198L;
-        private final int lastStreamId;
-        private final long errorCode;
-        private final byte[] debugData;
+        private GoAwayDetail goAwayDetail;
 
         public Http2GoAwayException(int lastStreamId, long errorCode, byte[] debugData) {
+            this(new GoAwayDetail(lastStreamId, errorCode, debugData));
+        }
+
+        private Http2GoAwayException(GoAwayDetail goAwayDetail) {
             super(Http2Error.STREAM_CLOSED);
-            this.lastStreamId = lastStreamId;
-            this.errorCode = errorCode;
-            this.debugData = debugData;
+            this.goAwayDetail = goAwayDetail;
         }
 
         public int lastStreamId() {
-            return lastStreamId;
+            return goAwayDetail.lastStreamId;
         }
 
         public long errorCode() {
-            return errorCode;
+            return goAwayDetail.errorCode;
         }
 
         public byte[] debugData() {
-            return debugData;
+            return goAwayDetail.debugData;
```

Comment:
I think Norman was expecting a clone() here as well, to guarantee the array isn't corrupted.